src/main/java/com/github/danielwegener/logback/kafka/KafkaAppender.java

```diff
@@ -46,7 +46,10 @@ public KafkaAppender() {
     public void doAppend(E e) {
         ensureDeferredAppends();
         if (e instanceof ILoggingEvent && ((ILoggingEvent)e).getLoggerName().startsWith(KAFKA_LOGGER_PREFIX)) {
-            deferAppend(e);
+            //only in case the producer is initialized we are able to send messages.
+            if ( (lazyProducer != null ) && lazyProducer.isInitialized() ) {
```

Comment:
I think this condition should rather guard the `ensureDeferredAppends();`in line 47, not? Buffering up messages in deferAppend while the producer is not initialized is somewhat intentional